File: files/CH-193.R
library(tidyverse)
library(readxl)

path = "files/CH-193 Custom Grouping.xlsx"
input = read_excel(path, range = "B2:C37")
test  = read_excel(path, range = "G2:H4")

result = input %>%
  mutate(Group = ifelse(wday(Date, week_start = 1) >= 6, "Weekend", "Weekday")) %>%
  summarise(`Total Sales` = sum(Sales), .by = Group)

all.equal(result, test, check.attributes = FALSE)
# [1] TRUE

File: files/100-199/193/CH-193.py
import pandas as pd

path = "CH-193 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=36)
test = pd.read_excel(path, usecols="G:H", skiprows=1, nrows=2)

result = input.assign(Group=input['Date'].dt.weekday >= 5).groupby('Group')['Sales'].sum().reset_index().rename(columns={'Group': 'Weekend', 'Sales': 'Total Sales'})

print(result.equals(test))